File: exercicios/Projeto Final/Projeto Back/Projeto/Model/Venda.cs
namespace Projeto.Model
{
    public class Venda
    {
        public int ID { get; set; }
        public int ProdutoID { get; set; }
        public int quantidade { get; set; }
        public int ClientID { get; set; }
        public double valor { get; set; }
        public double preco { get; set; }
        //public virtual Produto Produto { get; set; }
        //public virtual Cliente Cliente { get; set; }
    }
}

File: Projeto Angu/src/app/venda/venda.ts
import { Cliente } from "../cliente/cliente";
import { Produto } from "../produto/produto";

export class Venda{
    id?: number;
    produtoID?: number;
    quantidade?: number;
    clientID?: number;
    valor?: number;
    preco?: number;

    constructor(id?: number, cliente?: Cliente, produto?: Produto, quantidade?: number){
        this.id = id;
        this.clientID = cliente?.id;
        this.produtoID = produto?.id;
        this.quantidade = quantidade;
        this.preco = produto?.preco;
    }
}